reminder/database.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import annotations
+
+from datetime import datetime
+
+import enum
+from sqlalchemy import BigInteger, Column, Integer, String, DateTime, Enum
+from typing import List, Dict
+
+from database import database, session
+
+
+class ReminderStatus(enum.Enum):
+    WAITING: str = "WAITING"
+    REMINDED: str = "REMINDED"
+    FAILED: str = "FAILED"
+
+    def str_list() -> str:
+        return ", ".join([e.name for e in ReminderStatus])
+
+
+class ReminderItem(database.base):
+    """Represents a database Reminder item for :class:`Reminder` module.
+
+    Attributes:
+        idx: The database ID.
+        guild_id: ID of the guild.
+        author_id: User ID of reminder author.
+        remind_id: User ID of reminded user.
+        permalink: Message URL.
+        message: Reminder text (None if empty).
+        origin_date: Date of creation.
+        remind_date: Date for reminding
+        status: Status of reminder
+    """
+
+    __tablename__ = "reminder_reminder_reminderitem"
+
+    idx = Column(Integer, primary_key=True, autoincrement=True)
+    guild_id = Column(BigInteger, default=None)
+    author_id = Column(BigInteger, default=None)
+    remind_id = Column(BigInteger, default=None)
+    permalink = Column(String, default=None)
+    message = Column(String, default=None)
+    origin_date = Column(DateTime(timezone=True), default=None)
+    remind_date = Column(DateTime(timezone=True), default=None)
+    status = Column(Enum(ReminderStatus), default=ReminderStatus.WAITING)
+
+    @staticmethod
+    def add(
+        guild_id: int,
+        author_id: int,
+        remind_id: int,
+        permalink: str,
+        message: str,
+        origin_date: datetime,
+        remind_date: datetime,
+    ) -> ReminderItem:
+        """Creates a new ReminderItem in the database.
+
+        Args:
+            guild_id: ID of the guild.
+            author_id: User ID of reminder author.
+            remind_id: User ID of reminded author.
+            permalink: URL of reminder message.
+            message: Reminder text (None if empty).
+            origin_date: Date of creation.
+            remind_date: Date for reminding
+
+        Raises:
+            ValueError: End time already passed.
+
+        Returns:
+            The created database item.
+        """
+        origin_date = datetime.now()
+        if remind_date < origin_date:
+            raise ValueError
+
+        item = ReminderItem(
+            guild_id=guild_id,
+            author_id=author_id,
+            remind_id=remind_id,
+            permalink=permalink,
+            message=message,
+            origin_date=origin_date,
+            remind_date=remind_date,
+            status=ReminderStatus.WAITING,
+        )
+
+        session.add(item)
+        session.commit()
+        return item
+
+    @staticmethod
+    def get_all(
```

Comment:
I believe calling this method `get` makes more sense since you decide if you want to get all results depending on the parameters.

Also I would use discord objects here  too (not pointing that out in the future)

Reply:
It depends. If you return query.all(), then it should be get_all even if it returns one or none object.